connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -104,6 +81,33 @@ public DelegatingClassLoader(List<Path> pluginLocations) {
         this(pluginLocations, DelegatingClassLoader.class.getClassLoader());
     }
 
+    public Set<PluginSource> sources() {
```

Comment:
Since this is only used in one non-testing place right now we can leave it as-is, but I'm wondering if we might extract this logic into a common place in a subsequent PR.

I see that the description explains how this logic is dependent on the `ClassLoader` used to load plugins, but if that's the only dependency, would it be enough to add a `ClassLoader` parameter to this method (along with, presumably, a `List<Path>` for the to-be-scanned plugin locations)?

Reply:
The refactor was just a little bit more involved (as I had to figure out a new mocking strategy for newPluginClassLoader) but I think it's better now. I like the static implementation, and I really like how empty the DCL class is now. 

I'm not sure exactly where to put the function though, and just stuffed it into PluginUtils for now. Do you think this is more appropriate somewhere else?

Reply:
I think it's fine in `PluginUtils` if we plan on using it in multiple places. Otherwise 

I'm not a huge fan of how we're using the `ClassLoaderFactory` class right now, though. I can see the value for it in the `Plugins` class when both methods are used, but requiring an instance of it in `PluginUtils::pluginSources` seems like overkill since we don't need access to the `newDelegatingClassLoader` method.

Could we create a separate `PluginClassLoaderFactory` interface, have `ClassLoaderFactory` implement that interface, and change the signature of `PluginUtils::pluginSources` to accept an instance of that interface instead of a `ClassLoaderFactory`?

Also, it may be a little unclear to first-time readers why we have the separate `ClassLoaderFactory` class since it tracks no (instance or static) state and seems at first glance like all of its logic might be a better fit for the `PluginUtils` class. Can we document in that class that its purpose is to provide a layer of indirection for the purpose of easier mocking in tests?

Finally, we don't have to copy the visibility of the methods that we've extracted into the `ClassLoaderFactory` class; IMO both of those can and should be made `public`.

Reply:
> I think it's fine in PluginUtils if we plan on using it in multiple places.

Right now it's used in Plugins (main) and DelegatingClassLoaderTest (test). This will be used in the manifest generation script (main), without the surrounding `Plugins`, so i'll leave it as-is.

I've made the other changes you mentioned.